recipes/dfba/run_test.py

```diff
@@ -0,0 +1,101 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+
+"""
+Simulate anaerobic growth of _E. coli_ on glucose and xylose.
+
+: Organism : _Escherichia coli_ str. K-12 substr. MG1655
+: Model : http://bigg.ucsd.edu/models/iJR904
+
+"""
+
+from os.path import dirname, join
+
+from cobra.io import read_sbml_model
+
+from dfba import DfbaModel, ExchangeFlux, KineticVariable
+
+
+# DfbaModel instance initialized with cobra model
+fba_model = read_sbml_model(
+    join(dirname(__file__), "iJR904.xml.gz")
+)
+fba_model.solver = "glpk"
+dfba_model = DfbaModel(fba_model)
+
+# instances of KineticVariable (default initial conditions are 0.0, but can be
+# set here if wanted e.g. Oxygen)
+X = KineticVariable("Biomass")
+Gluc = KineticVariable("Glucose")
+Xyl = KineticVariable("Xylose")
+Oxy = KineticVariable("Oxygen", initial_condition=0.24)
+Eth = KineticVariable("Ethanol")
+
+# add kinetic variables to dfba_model
+dfba_model.add_kinetic_variables([X, Gluc, Xyl, Oxy, Eth])
+
+# instances of ExchangeFlux
+mu = ExchangeFlux("BiomassEcoli")
+v_G = ExchangeFlux("EX_glc(e)")
+v_Z = ExchangeFlux("EX_xyl_D(e)")
+v_O = ExchangeFlux("EX_o2(e)")
+v_E = ExchangeFlux("EX_etoh(e)")
+
+# add exchange fluxes to dfba_model
+dfba_model.add_exchange_fluxes([mu, v_G, v_Z, v_O, v_E])
+
+# add rhs expressions for kinetic variables in dfba_model
+dfba_model.add_rhs_expression("Biomass", mu * X)
+dfba_model.add_rhs_expression("Glucose", v_G * 180.1559 * X / 1000.0)
+dfba_model.add_rhs_expression("Xylose", v_Z * 150.13 * X / 1000.0)
+dfba_model.add_rhs_expression("Oxygen", v_O * 16.0 * X / 1000.0)
+dfba_model.add_rhs_expression("Ethanol", v_E * 46.06844 * X / 1000.0)
+
+# add lower/upper bound expressions for exchange fluxes in dfba_model together
+# with expression that must be non-negative for correct evaluation of bounds
+dfba_model.add_exchange_flux_lb(
+    "EX_glc(e)", 10.5 * (Gluc / (0.0027 + Gluc)) * (1 / (1 + Eth / 20.0)), Gluc
+)
+dfba_model.add_exchange_flux_lb("EX_o2(e)", 15.0 * (Oxy / (0.024 + Oxy)), Oxy)
+dfba_model.add_exchange_flux_lb(
+    "EX_xyl_D(e)",
+    6.0 * (Xyl / (0.0165 + Xyl)) * (1 / (1 + Eth / 20.0)) * (1 / (1 + Gluc / 0.005)),
+    Xyl,
+)
+
+# add initial conditions for kinetic variables in dfba_model biomass (gDW/L),
+# metabolites (g/L)
+dfba_model.add_initial_conditions(
+    {
+        "Biomass": 0.03,
+        "Glucose": 15.5,
+        "Xylose": 8.0,
+        "Oxygen": 0.0,
+        "Ethanol": 0.0,
+    }
+)
+
+# simulate model across interval t = [0.0,25.0](hours) with outputs for plotting
+# every 0.1h and optional list of fluxes
+concentrations, trajectories = dfba_model.simulate(
+    0.0, 25.0, 0.1, ["EX_glc(e)", "EX_xyl_D(e)", "EX_etoh(e)"]
+)
+
+# TODO: Assert some value ranges for the results.
```

Comment:
Did you want to do anything about this "TODO"?

Reply:
Not at the moment. I'm happy with knowing that the simulation completes successfully.